presto-lark-sheets/src/main/java/com/facebook/presto/lark/sheets/SheetsMetadata.java

```diff
@@ -0,0 +1,343 @@
+    }
+
+    @Override
+    public void createSchema(ConnectorSession session, String schemaName, Map<String, Object> properties)
+    {
+        String token = SheetsSchemaProperties.getSchemaToken(properties).orElseThrow(
+                () -> new PrestoException(SCHEMA_TOKEN_NOT_PROVIDED, "Schema token is required but not provided"));
+        boolean publicVisible = SheetsSchemaProperties.isSchemaPublic(properties);
+        schemaStore.insert(new SheetsSchema(schemaName, session.getUser(), token, publicVisible));
+    }
+
+    @Override
+    public void dropSchema(ConnectorSession session, String schemaName)
+    {
+        SheetsSchema schema = requireVisibleSchema(session, schemaName);
+        checkSchemaUpdatable(schema, session.getUser(), "drop");
+        schemaStore.delete(schemaName);
+    }
+
+    @Override
+    public void renameSchema(ConnectorSession session, String source, String target)
+    {
+        SheetsSchema schema = requireVisibleSchema(session, source);
+        checkSchemaUpdatable(schema, session.getUser(), "rename");
+        schemaStore.rename(source, target);
+    }
+
+    private Optional<SheetsSchema> getVisibleSchema(ConnectorSession session, String schemaName)
+    {
+        return schemaStore.get(schemaName)
+                .filter(schema -> schema.isVisibleTo(session.getUser()));
+    }
+
+    private SheetsSchema requireVisibleSchema(ConnectorSession session, String schemaName)
+    {
+        return getVisibleSchema(session, schemaName)
+                .orElseThrow(() -> new PrestoException(SCHEMA_NOT_EXISTS,
+                        format("Schema %s not exists or not visible", schemaName)));
+    }
+
+    private void checkSchemaUpdatable(SheetsSchema schema, String operationUser, String operation)
+    {
+        if (!schema.getUser().equalsIgnoreCase(operationUser)) {
+            throw new PrestoException(NOT_PERMITTED,
+                    format("User '%s' is not permitted to perform '%s' on schema '%s'", operationUser, operation, schema.getName()));
+        }
+    }
+
+    private void checkSchemaReadable(SheetsSchema schema)
+    {
+        if (!sheetsApi.isReadable(schema.getToken())) {
+            throw new PrestoException(SCHEMA_NOT_READABLE,
+                    format("Spreadsheet %s not readable", schema.getToken()));
+        }
+    }
+
+    private List<SheetsColumnHandle> getColumns(SheetsTableHandle table)
+    {
+        List<String> header = sheetsApi.getHeaderRow(table.getSpreadsheetToken(), table.getSheetId(), table.getColumnCount());
+
+        int numColumns = header.size();
+        LinkedHashMap<String, SheetsColumnHandle> columns = new LinkedHashMap<>(numColumns);
+        for (int i = 0; i < numColumns; i++) {
+            String rawColumnName = header.get(i);
+            if (rawColumnName == null) {
+                // Columns without name are ignored
+                continue;
+            }
+            String columnName = rawColumnName.toLowerCase(ENGLISH);
```

Comment:
What if the column name is non-English? Will it throw an exception or just skip the column silently?

Reply:
Presto supports UTF-8 identifier names. So for a non-english string, it can be used as a column name in Lark connector, but should be quoted as `"XXX"`to access the column.

Reply:
Gotcha. Thanks!